source/server/http/admin.cc

```diff
@@ -402,11 +404,21 @@ Http::Code AdminImpl::handlerStats(absl::string_view url, Http::HeaderMap& respo
     all_stats.emplace(gauge->name(), gauge->value());
   }
 
+  for (const Stats::HistogramSharedPtr& histogram : server_.stats().histograms()) {
+    all_histograms.emplace(histogram->name(),
+                           fmt::format("\n\t Interval:   {}\n\t Cumulative: {}",
+                                       histogram->intervalStatistics().summary(),
+                                       histogram->cumulativeStatistics().summary()));
+  }
+
   if (params.size() == 0) {
     // No Arguments so use the standard.
     for (auto stat : all_stats) {
       response.add(fmt::format("{}: {}\n", stat.first, stat.second));
     }
+    for (auto histogram : all_histograms) {
```

Comment:
It looks like all_histograms is written but not read when params is non-empty.  If that's desired, then sink the population of all_histograms into this branch of the if.

However, I wonder if we want to also output the histogram data for 'format=json'.  WDYT?

Reply:
+1 we should definitely do for format=json. I actually want to actually think about what the output format looks like in both cases. @ramaraochavali can you add an example of what it currently looks like so we can discuss?

Reply:
This is in my TODO list. But did not bring this up in this PR to avoid review churn. I will add it.

Reply:
Here is the format in /stats end point
cluster.time.external.upstream_rq_time: 
	 Interval:   P0: 0, P25: 0, P50: 0, P75: 1.01284, P90: 1.06981, P95: 1.0888, P99: 2.05997, P99.9: 4.02908, P100: 4.1
	 Cumulative: P0: 0, P25: 0, P50: 0, P75: 1.00037, P90: 1.06434, P95: 1.08567, P99: 2.04541, P99.9: 4.02094, P100: 14

Here is the format in /stats?format=json end point
        {
            "name": "cluster.time.external.upstream_rq_time",
            "quantiles": [
                {
                    "quantile": 0.0,
                    "interval_value": 0.0,
                    "cumulative_value": 0.0
                },
                {
                    "quantile": 25.0,
                    "interval_value": 0.0,
                    "cumulative_value": 0.0
                },
                {
                    "quantile": 50.0,
                    "interval_value": 0.0,
                    "cumulative_value": 0.0
                },
                {
                    "quantile": 75.0,
                    "interval_value": 1.0128384409401963,
                    "cumulative_value": 1.000373952931791
                },
                {
                    "quantile": 90.0,
                    "interval_value": 1.0698125557869683,
                    "cumulative_value": 1.0643438372556842
                },
                {
                    "quantile": 95.0,
                    "interval_value": 1.0888039274025588,
                    "cumulative_value": 1.085667132030315
                },
                {
                    "quantile": 99.0,
                    "interval_value": 2.0599732142857145,
                    "cumulative_value": 2.0454053156146174
                },
                {
                    "quantile": 99.9,
                    "interval_value": 4.02907777777778,
                    "cumulative_value": 4.0209359999999869
                },
                {
                    "quantile": 100.0,
                    "interval_value": 4.1,
                    "cumulative_value": 14.0
                }
            ]
        },

Please note that if a particular histogram is not used in a given interval, all the interval quantile values would be Nan in case of /stats end point and null in case of /stats?format=json

Reply:
I wonder if NaN is the easiest-to-use sentinel here, as json gets parsed by multiple languages.  Would we lose generality if we used a sentinel of -1?  Or is NaN a pretty common sentinel in json?

Reply:
Json with Nan is not working - it is not getting displayed either via browser or through curl. That is why I skipped setting value if it is Nan and hence it returns null which is fine I think.

Reply:
Yes it works.

Reply:
@ramaraochavali my concern about the text output version is that it is not good for 1-line grep, which I think is a common use case (at least it is for me). It might be ugly, but can we figure out a format for the text version which has everything on a single line so we can grep for the stat?

Reply:
@mattklein123 I tried to compact it to a single line. It looks like this
`cluster.time.upstream_rq_time: P0(0,0) P25(0,0) P50(0,0) P75(0,0) P90(1.04532,0) P95(1.07494,1.00249) P99(1.09864,1.08607) P99.9(6.03333,2.08273) P100(8.1,53)`
First value is interval value and second value is cumulative value. I think we can add documentation line and it should be clear. what do you think?

Reply:
this format LGTM; easy to parse with a shell script if needed.